kong/db/strategies/cassandra/init.lua

```diff
@@ -1203,7 +1202,15 @@ do
   end
 
   function _mt:page(size, offset, options, foreign_key, foreign_key_db_columns)
-    size = size or constants.DEFAULT_PAGE_SIZE
+    if not size then
+      if type(options) == "table" and type(options.pagination) == "table" then
+        size = options.pagination.page_size
+      end
+
+      if not size then
+        size = self.connector.defaults.pagination.page_size
+      end
+    end
```

Comment:
likewise, this is being added in lots of places... I wonder if there's a way to make future maintenance of this logic easier to ensure it remains consistent. How about

```lua
size = size or self.connector:get_page_size(options)
```

with connector having something like

```lua
function connector:get_page_size(options)
  return (options and options.pagination)
         and options.pagination.page_size
         or  self.defaults.pagination.page_size
end
```

Reply:
I changed this.